### PR TITLE
Fix absolute paths to compiler/ directories in CODEOWNERS.

### PR DESCRIPTION
**Pull request by ScottTodd**
Missed this when reviewing https://github.com/google/iree/pull/9022, but these paths are absolute from the root of the repo, so just `/compiler/` won't trigger as expected.

Also add Ben back as the top level default (from back when `/iree/` covered `/iree/compiler/`)

### CODE REVIEW

**Comment by MaheshRavishankar** [comment]
Github is complaining about this file as having errors. Do you see that here https://github.com/google/iree/pull/9054/files# . It seems I cannot approve it because of the errors.

**Comment by ScottTodd** [comment]
> Github is complaining about this file as having errors. Do you see that here https://github.com/google/iree/pull/9054/files# . It seems I cannot approve it because of the errors.

Geoffrey set that up and it works as expected. See the comment at the top of the file: https://github.com/google/iree/blob/e2f6c0f1deb47bee7c60e7ab37f771e935252f66/.github/CODEOWNERS#L5-L7

**Review by MaheshRavishankar — APPROVED** [review]
My bad. Somehow the review changes button wasnt showing up.

**Pull request merged**
